inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -336,8 +273,9 @@ public function is_valid_post_type( $post_type ) {
 	protected function get_excluded_posts( $post_type ) {
 		$excluded_posts_ids = array();
 
-		if ( $post_type === 'page' && $this->get_page_for_posts_id() ) {
-			$excluded_posts_ids[] = $this->get_page_for_posts_id();
+		$page_on_front_id = ( $post_type === 'page' ) ? (int) get_option( 'page_on_front' ) : 0;
```

Comment:
I think we can simplify this further once https://github.com/Yoast/wordpress-seo/pull/13075 is merged, because we needn't have this logic in a ton of different places :)